core/src/main/java/io/grpc/CallOptions.java

```diff
@@ -363,6 +371,70 @@ public boolean isWaitForReady() {
   }
 
   /**
+   * Sets the maximum allowed message size acceptable from the remote peer.  If unset, this will
+   * default to the value set on the {@link ManagedChannel}.
+   */
+  @ExperimentalApi("https://github.com/grpc/grpc-java/issues/FIXME")
+  public CallOptions withMaxInboundMessageSize(int maxSize) {
+    checkArgument(maxSize >= 0, "invalid maxsize %s", maxSize);
+    CallOptions newOptions = new CallOptions(this);
+    newOptions.maxInboundMessageSize = maxSize;
+    return newOptions;
+  }
+
+  /**
+   * Sets the maximum allowed message size acceptable sent to the remote peer.  If unset, this will
+   * default to the value set on the {@link ManagedChannel}.
+   */
+  @ExperimentalApi("https://github.com/grpc/grpc-java/issues/FIXME")
+  public CallOptions withMaxOutboundMessageSize(int maxSize) {
+    checkArgument(maxSize >= 0, "invalid maxsize %s", maxSize);
+    CallOptions newOptions = new CallOptions(this);
+    newOptions.maxOutboundMessageSize = maxSize;
+    return newOptions;
+  }
+
+  /**
+   * Clears the maximum allowed message size acceptable from the remote peer.  The value set on the
+   * {@link ManagedChannel} may still be enforced.
+   */
+  @ExperimentalApi("https://github.com/grpc/grpc-java/issues/FIXME")
+  public CallOptions withoutMaxInboundMessageSize() {
```

Comment:
There are not two ways.  There is the primitive way of setting the max, and the clearing without method.  There is no `Integer` way of clearing.  This is consistent with `withWaitForReady` and `withoutWaitForReady`.  You imply that that should be `withWaitForReady(Boolean)`.

It's also Effective Java Item 49 says Prefer primitives to Boxed.

Creating lots of stubs would have a memory problem, but unboxing the integer on every call seems unnecessary.